File: pydss/control/ControlClassImpl.py
from dss.control.impl.ControlElemImpl import (ControlElemImpl,)
from dss.common.impl.CktElementClassImpl import (CktElementClassImpl,)
from dss.common.impl.DSSClassDefs import (DSSClassDefs,)
from dss.control.ControlClass import (ControlClass,)


class ControlClassImpl(CktElementClassImpl, ControlClass):
    numControlClassProps = None

    def __init__(self):
        super(ControlClassImpl, self)()
        self.numControlClassProps = 0
        self.classType = DSSClassDefs.CTRL_ELEMENT

    def countProperties(self):
        self.numProperties = self.numProperties + self.numControlClassProps
        super(ControlClassImpl, self).countProperties()

    def defineProperties(self):
        self.activeProperty = self.activeProperty + self.numControlClassProps
        super(ControlClassImpl, self).defineProperties()

    def classEdit(self, activeControlObj, paramPointer):
        if paramPointer >= 0:
            super(ControlClassImpl, self).classEdit(activeControlObj, paramPointer - self.numControlClassProps)
        return 0

    def classMakeLike(self, otherObj):
        ControlElemImpl(otherObj)

    def getNumControlClassProps(self):
        return self.numControlClassProps

    def setNumControlClassProps(self, num):
        self.numControlClassProps = num

File: core/dss-core/src/main/java/com/ncond/dss/control/impl/ControlClassImpl.java
package com.ncond.dss.control.impl;

import com.ncond.dss.common.DSSClass;
import com.ncond.dss.common.impl.CktElementClassImpl;
import com.ncond.dss.common.impl.DSSClassDefs;
import com.ncond.dss.control.ControlClass;
import com.ncond.dss.general.DSSObject;

public abstract class ControlClassImpl extends CktElementClassImpl implements ControlClass {

	private int numControlClassProps;

	public ControlClassImpl() {
		super();
		numControlClassProps = 0;
		classType = DSSClassDefs.CTRL_ELEMENT;
	}

	@Override
	protected void countProperties() {
		numProperties = numProperties + numControlClassProps;
		super.countProperties();
	}

	@Override
	protected void defineProperties() {
		activeProperty = activeProperty + numControlClassProps;

		super.defineProperties();
	}

	@Override
	protected int classEdit(DSSObject activeControlObj, int paramPointer) {

		if (paramPointer >= 0)
			super.classEdit(activeControlObj, paramPointer - numControlClassProps);

		return 0;
	}

	protected void classMakeLike(DSSClass otherObj) {
//		new ControlElemImpl(otherObj);
	}

	@Override
	public int getNumControlClassProps() {
		return numControlClassProps;
	}

	@Override
	public void setNumControlClassProps(int num) {
		numControlClassProps = num;
	}

}
